sklearn/linear_model/tests/test_logistic.py

```diff
@@ -801,26 +802,19 @@ def test_logistic_regression_solvers():
     X, y = make_classification(n_features=10, n_informative=5, random_state=0)
 
     params = dict(fit_intercept=False, random_state=42, multi_class="ovr")
-    ncg = LogisticRegression(solver="newton-cg", **params)
-    lbf = LogisticRegression(solver="lbfgs", **params)
-    lib = LogisticRegression(solver="liblinear", **params)
-    sag = LogisticRegression(solver="sag", **params)
-    saga = LogisticRegression(solver="saga", **params)
-    ncg.fit(X, y)
-    lbf.fit(X, y)
-    sag.fit(X, y)
-    saga.fit(X, y)
-    lib.fit(X, y)
-    assert_array_almost_equal(ncg.coef_, lib.coef_, decimal=3)
-    assert_array_almost_equal(lib.coef_, lbf.coef_, decimal=3)
-    assert_array_almost_equal(ncg.coef_, lbf.coef_, decimal=3)
-    assert_array_almost_equal(sag.coef_, lib.coef_, decimal=3)
-    assert_array_almost_equal(sag.coef_, ncg.coef_, decimal=3)
-    assert_array_almost_equal(sag.coef_, lbf.coef_, decimal=3)
-    assert_array_almost_equal(saga.coef_, sag.coef_, decimal=3)
-    assert_array_almost_equal(saga.coef_, lbf.coef_, decimal=3)
-    assert_array_almost_equal(saga.coef_, ncg.coef_, decimal=3)
-    assert_array_almost_equal(saga.coef_, lib.coef_, decimal=3)
+    solvers = ("newton-cg", "lbfgs", "liblinear", "sag", "saga", "trust-ncg")
+
+    # dict of fitted logistic regressors
+    regressors = {
+        solver: LogisticRegression(solver=solver, **params).fit(X, y)
+        for solver in solvers
+    }
+
+    # test over unordered pairs of distinct solvers
+    for solver_1, solver_2 in itertools.combinations(regressors):
```

Comment:
We do not need to test every single combination. We can have a list of fitted estimators and compare the coefs of adjacent pairs.

Reply:
I was mainly worried about small errors accumulating in such a chain so that it seems like transitivity holds while two sufficiently distant solvers don't actually converge to the same result. If you think that's a negligible worry, then I can change the implementation. I just think its a _tiny bit_ safer as is.

Reply:
Okay, let's stick with combinations and see what others think.

Reply:
Actually, I suppose I can just set `atol` sufficiently low for the check. Will update.